File: src/main/java/com/lord/small_box/models/PurchaseOrder.java
package com.lord.small_box.models;

import java.math.BigDecimal;
import java.util.Calendar;
import java.util.List;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
@Table(name = "purchase_order")
public class PurchaseOrder {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;

	private int orderNumber;

	private String jurisdiction;
	
	@ManyToOne(cascade = CascadeType.MERGE,fetch = FetchType.LAZY)
	@JoinColumn(name="organization_id",referencedColumnName = "id")
	private Organization organization;

	private String executerUnit;

	private String financingSource;

	private String dependency;

	private String provider;

	private String deliverTo;

	private Calendar date;

	private String exp;
	
	private BigDecimal purchaseOrderTotal;
	
	private boolean loadedToDeposit;

	
	

}

File: front/small-box/src/app/models/purchaseOrder.ts
import { PurchaseOrderItemDto } from "./PurchaseOrderItemDto";

export class PurchaseOrder{
    id!:number;
    orderNumber!:number;
    jurisdiction!:string;
    executerUnit!:string;
    executerUnitOrganizationId!:number;
    financingSource!:string;
    dependency!:string;
    dependencyOrganizacionId!:number;
    provider!:string;
    deliverTo!:string;
    date!:Date;
    exp!:string;
    purchaseOrderTotal!:number;
    items!:Array<PurchaseOrderItemDto>;
}